tests/memoryview/numpy_memoryview.pyx

```diff
@@ -718,3 +718,41 @@ def test_boundscheck_and_wraparound(double[:, :] x):
         x[i]
         x[i, ...]
         x[i, :]
+
+
+ctypedef struct SameTypeAfterArraysStructSimple:
+    double a[16]
+    double b[16]
+    double c
+
+@testcase
+def same_type_after_arrays_simple():
+    """
+    >>> same_type_after_arrays_simple()
+    """
+
+    cdef SameTypeAfterArraysStructSimple element
+    arr = np.ones(2, np.asarray(<SameTypeAfterArraysStructSimple[:1]>&element).dtype)
```

Comment:
There are also more thorough and low-level buffer format tests in `tests/buffers/buffmt.pyx` that do not require NumPy to run, and thus are preferable for this kind of parsing test.

The tests here are still good as integration tests, though, so keep them.

Reply:
I'll take a look at this now and implement some tests...